src/components/KYCWall/BaseKYCWall.js

```diff
@@ -17,75 +17,92 @@ import * as ReportUtils from '../../libs/ReportUtils';
 // This component allows us to block various actions by forcing the user to first add a default payment method and successfully make it through our Know Your Customer flow
 // before continuing to take whatever action they originally intended to take. It requires a button as a child and a native event so we can get the coordinates and use it
 // to render the AddPaymentMethodMenu in the correct location.
-class KYCWall extends React.Component {
-    constructor(props) {
-        super(props);
-
-        this.continue = this.continue.bind(this);
-        this.setMenuPosition = this.setMenuPosition.bind(this);
-        this.anchorRef = React.createRef(null);
-
-        this.state = {
-            shouldShowAddPaymentMenu: false,
-            anchorPositionVertical: 0,
-            anchorPositionHorizontal: 0,
-            transferBalanceButton: null,
-        };
-    }
-
-    componentDidMount() {
-        PaymentMethods.kycWallRef.current = this;
-        if (this.props.shouldListenForResize) {
-            this.dimensionsSubscription = Dimensions.addEventListener('change', this.setMenuPosition);
-        }
-        Wallet.setKYCWallSourceChatReportID(this.props.chatReportID);
-    }
+function KYCWall({
+    shouldListenForResize,
+    chatReportID,
+    popoverPlacement,
+    iouReport,
+    fundList,
+    reimbursementAccount,
+    bankAccountList,
+    userWallet,
+    enablePaymentsRoute,
+    onSuccessfulKYC,
+    addBankAccountRoute,
+    addDebitCardRoute,
+    children,
+}) {
+    const anchorRef = useRef(null);
+    const transferBalanceButtonRef = useRef(null);
 
-    componentWillUnmount() {
-        if (this.props.shouldListenForResize && this.dimensionsSubscription) {
-            this.dimensionsSubscription.remove();
-        }
-        PaymentMethods.kycWallRef.current = null;
-    }
-
-    setMenuPosition() {
-        if (!this.state.transferBalanceButton) {
-            return;
-        }
-        const buttonPosition = getClickedTargetLocation(this.state.transferBalanceButton);
-        const position = this.getAnchorPosition(buttonPosition);
-        this.setPositionAddPaymentMenu(position);
-    }
+    const [shouldShowAddPaymentMenu, setShouldShowAddPaymentMenu] = useState(false);
+    const [anchorPosition, setAnchorPosition] = useState({
+        anchorPositionVertical: 0,
+        anchorPositionHorizontal: 0,
+    });
 
     /**
      * @param {DOMRect} domRect
      * @returns {Object}
      */
-    getAnchorPosition(domRect) {
-        if (this.props.popoverPlacement === 'bottom') {
+    const getAnchorPosition = useCallback(
+        (domRect) => {
+            if (popoverPlacement === 'bottom') {
+                return {
+                    anchorPositionVertical: domRect.top + (domRect.height - CONST.MODAL.POPOVER_MENU_ANCHOR_POSITION_VERTICAL_OFFSET),
```

Comment:
Sorry, should've been more specific. I'd create a const before component declaration, these values are only being used in this component. Dont think there is a need to put them in global CONST file